contracts/modules/TransactionManager.sol

```diff
@@ -0,0 +1,273 @@
+    bytes4 private constant ERC721_RECEIVED = bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"));
+    bytes4 private constant ERC1155_RECEIVED = bytes4(keccak256("onERC1155Received(address,address,uint256,uint256,bytes)"));
+    bytes4 private constant ERC1155_BATCH_RECEIVED = bytes4(keccak256("onERC1155BatchReceived(address,address,uint256[],uint256[],bytes)"));
+    bytes4 private constant ERC165_INTERFACE = bytes4(keccak256("supportsInterface(bytes4)"));
+    bytes4 private constant ERC1155_INTERFACE = ERC1155_RECEIVED ^ ERC1155_BATCH_RECEIVED;
+
+    struct Call {
+        address to;
+        uint256 value;
+        bytes data;
+        bool isSpenderInData;
+    }
+
+    // *************** Events *************************** //
+
+    event Transfer(address indexed wallet, address indexed token, uint256 indexed amount, address to, bytes data);
+    event CalledContract(address indexed wallet, address indexed to, uint256 amount, bytes data);
+    event AddedToWhitelist(address indexed wallet, address indexed target, uint64 whitelistAfter);
+    event RemovedFromWhitelist(address indexed wallet, address indexed target);
+    event ToggledDappRegistry(address _wallet, bytes32 _registry, bool isEnabled);
+    event SessionCreated(address indexed wallet, address sessionKey, uint64 expires);
+    event SessionCleared(address indexed wallet, address sessionKey);
+    event ERC1155TokenReceiverEnabled(address indexed _wallet);
+
+    // *************** External functions ************************ //
+
+    function multiCall(
+        address _wallet,
+        Call[] calldata _transactions
+    )
+        external
+        onlySelf()
+        onlyWhenUnlocked(_wallet)
+        returns (bytes[] memory)
+    {
+        bytes[] memory results = new bytes[](_transactions.length);
+        for(uint i = 0; i < _transactions.length; i++) {
+            address spender = recoverSpender(_wallet, _transactions[i]);
+            require(
+                isWhitelisted(_wallet, spender) ||
+                isAuthorised(_wallet, spender, _transactions[i].to, _transactions[i].data), "TM: call not authorised");
+            results[i] = invokeWallet(_wallet, _transactions[i].to, _transactions[i].value, _transactions[i].data);
+        }
+        return results;
+    }
+
+    function multiCallWithSession(
+        address _wallet,
+        Call[] calldata _transactions
+    )
+        external
+        onlySelf()
+        onlyWhenUnlocked(_wallet)
+        returns (bytes[] memory)
+    {
+        multiCallWithApproval(_wallet, _transactions);
+    }
+
+    function multiCallWithGuardians(
+        address _wallet,
+        Call[] calldata _transactions
+    )
+        external 
+        onlySelf()
+        onlyWhenUnlocked(_wallet)
+        returns (bytes[] memory)
+    {
+        multiCallWithApproval(_wallet, _transactions);
+    }
+
+    function multiCallWithGuardiansAndStartSession(
```

Comment:
StartingSession instead of AndStartSession?

"And" feels like session happens after (and order not relevant)